wrangler-storage/src/main/java/co/cask/wrangler/dataset/workspace/WorkspaceDataset.java

```diff
@@ -186,50 +143,83 @@ public boolean hasWorkspace(String id) throws WorkspaceException {
-      );
     }
     return values;
   }
 
-  private boolean excludedKey(String id) {
-    return id.equalsIgnoreCase(Bytes.toString(CONFIG_KEY));
+  /**
+   * Update the properties of the specified workspace.
+   *
+   * @param id the workspace id
+   * @param properties the properties to update
+   * @throws WorkspaceNotFoundException if the workspace does not exist
+   */
+  public void updateWorkspaceProperties(String id, Map<String, String> properties) throws WorkspaceNotFoundException {
+    Workspace existing = getWorkspace(id);
+    Workspace updated = Workspace.builder(existing)
+      .setProperties(properties)
+      .setUpdated(System.currentTimeMillis() / 1000)
+      .build();
+    table.put(toPut(updated));
+  }
+
+  /**
+   * Update the directive execution request for the specified workspace.
+   *
+   * @param id the workspace id
+   * @param request the directive execution request
+   * @throws WorkspaceNotFoundException if the workspace does not exist
+   */
+  public void updateWorkspaceRequest(String id, Request request) throws WorkspaceNotFoundException {
+    Workspace existing = getWorkspace(id);
+    Workspace updated = Workspace.builder(existing)
+      .setRequest(request)
+      .setUpdated(System.currentTimeMillis() / 1000)
+      .build();
+    table.put(toPut(updated));
+  }
+
+  /**
+   * Update the sample data for the specified workspace.
+   *
+   * @param id the workspace id
+   * @param data the sample data
+   * @throws WorkspaceNotFoundException if the workspace does not exist
+   */
+  public void updateWorkspaceData(String id, DataType dataType, byte[] data) throws WorkspaceNotFoundException {
+    Workspace existing = getWorkspace(id);
+    Workspace updated = Workspace.builder(existing)
+      .setType(dataType)
+      .setData(data)
+      .setUpdated(System.currentTimeMillis() / 1000)
+      .build();
+    table.put(toPut(updated));
   }
 
   /**
    * Deletes the workspace.
    *
    * @param id to be deleted.
-   * @throws WorkspaceException thrown if there is issue deleting workspace.
    */
-  @WriteOnly
-  public void deleteWorkspace(String id) throws WorkspaceException {
-    try {
-      table.delete(toKey(id));
-    } catch (DataSetException e) {
-      throw new WorkspaceException(
-        String.format("Failed to delete workspace '%s'. %s", id, e.getMessage())
-      );
-    }
+  public void deleteWorkspace(String id) {
+    table.delete(Bytes.toBytes(id));
   }
 
   /**
    * Deletes a group of workspaces.
    *
+   * TODO: (CDAP-14692) make sure scope is indexed so this doesn't require a full table scan
+   *
    * @param group to be deleted.
    * @return number of workspaces deleted within a group.
-   * @throws WorkspaceException thrown if there are issues with deleting workspace within group.
    */
-  @WriteOnly
-  public int deleteGroup(String group) throws WorkspaceException {
+  public int deleteGroup(String group) {
```

Comment:
is a group the same as a scope? If so, maybe name the method and parameters that way

Reply:
done